File: mns-core/src/test/java/cz/fb/manaus/core/model/PriceComparatorTest.java
package cz.fb.manaus.core.model;

import org.junit.Test;

import java.util.List;

import static cz.fb.manaus.core.model.PriceComparator.ORDERING;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class PriceComparatorTest {

    @Test
    public void testComparisonLay() throws Exception {
        assertTrue(ORDERING.isOrdered(List.of(new Price(2d, 5d, Side.LAY), new Price(2.1d, 5d, Side.LAY))));
        assertTrue(ORDERING.isOrdered(List.of(new Price(2d, 5d, Side.LAY), new Price(2d, 5d, Side.LAY))));
        assertFalse(ORDERING.isOrdered(List.of(new Price(2.1d, 5d, Side.LAY), new Price(2d, 5d, Side.LAY))));
    }

    @Test
    public void testComparisonBack() throws Exception {
        assertTrue(ORDERING.isOrdered(List.of(new Price(2.1d, 5d, Side.BACK), new Price(2d, 5d, Side.BACK))));
        assertTrue(ORDERING.isOrdered(List.of(new Price(2d, 5d, Side.BACK), new Price(2d, 5d, Side.BACK))));
        assertFalse(ORDERING.isOrdered(List.of(new Price(2d, 5d, Side.BACK), new Price(2.1d, 5d, Side.BACK))));
    }

    @Test
    public void testComparison() throws Exception {
        assertTrue(ORDERING.isOrdered(List.of(new Price(2d, 5d, Side.BACK), new Price(2d, 5d, Side.LAY))));
        assertFalse(ORDERING.isOrdered(List.of(new Price(2d, 5d, Side.LAY), new Price(2d, 5d, Side.BACK))));
    }


}

File: mns-core/src/test/java/cz/fb/manaus/core/model/PriceComparatorTest.kt
package cz.fb.manaus.core.model

import com.google.common.collect.Comparators
import org.junit.jupiter.api.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue


class PriceComparatorTest {

    private fun isSorted(l: List<Price>): Boolean {
        return Comparators.isInOrder(l, PriceComparator)
    }

    @Test
    fun testComparisonLay() {
        assertTrue(isSorted(listOf(Price(2.0, 5.0, Side.LAY), Price(2.1, 5.0, Side.LAY))))
        assertTrue(isSorted(listOf(Price(2.0, 5.0, Side.LAY), Price(2.1, 5.0, Side.LAY))))
        assertTrue(isSorted(listOf(Price(2.0, 5.0, Side.LAY), Price(2.0, 5.0, Side.LAY))))
        assertFalse(isSorted(listOf(Price(2.1, 5.0, Side.LAY), Price(2.0, 5.0, Side.LAY))))
    }

    @Test
    fun testComparisonBack() {
        assertTrue(isSorted(listOf(Price(2.1, 5.0, Side.BACK), Price(2.0, 5.0, Side.BACK))))
        assertTrue(isSorted(listOf(Price(2.0, 5.0, Side.BACK), Price(2.0, 5.0, Side.BACK))))
        assertFalse(isSorted(listOf(Price(2.0, 5.0, Side.BACK), Price(2.1, 5.0, Side.BACK))))
    }

    @Test
    fun testComparison() {
        assertTrue(isSorted(listOf(Price(2.0, 5.0, Side.BACK), Price(2.0, 5.0, Side.LAY))))
        assertFalse(isSorted(listOf(Price(2.0, 5.0, Side.LAY), Price(2.0, 5.0, Side.BACK))))
    }
}


